### PR TITLE
Notes about memory usage (helps with #2535)

### PR DESCRIPTION
**Pull request by brianshumate**

### CODE REVIEW

File: website/source/docs/guides/performance.html.markdown [hunk 1]
```diff
@@ -87,3 +87,20 @@ respect them.
 * In other applications that perform high volumes of reads against Consul, consider using the
 [stale consistency mode](/docs/agent/http.html#consistency) available to allow reads to scale
 across all the servers and not just be forwarded to the leader.
+
+<a name="memory-requirements"></a>
```
Comment (sethvargo) [inline website/source/docs/guides/performance.html.markdown]: You don't need this - middleman will automatically create this from the h2

File: website/source/docs/guides/performance.html.markdown [hunk 2]
```diff
@@ -87,3 +87,20 @@ respect them.
...
+
+Consul server agents operate on a working set of data comprised of key/value
+entries, the service catalog, prepared queries, access control lists, and
+sessions in memory. These data are persisted through Raft to disk in the form
+of a snapshot and log of changes since the previous snapshot for durability.
+
+When planning for memory requirements, you should typically allocate
+enough RAM for your server agents to contain between 2 to 4 times the working
+set size. You can determine the working set size by noting the value of
+`consul.runtime.alloc_bytes` in the [Telemetry data](https://www.consul.io/docs/agent/telemetry.html).
```
Comment (sethvargo) [inline website/source/docs/guides/performance.html.markdown]: Do not use an absolute URL as it breaks local dev. This should be /docs/agent/telemetry.html instead.

**Comment by brianshumate** [comment]
Thanks for the tips @sethvargo!

**Comment by slackpad** [comment]
Thanks sorry I missed this one!

**Pull request merged**